src/task.rs

```diff
@@ -472,6 +472,9 @@ pub mod watchdog {
                 })?;
             }
 
+            #[cfg(esp_idf_version = "5.0")]
+            esp!(unsafe { esp_task_wdt_init(&config.into() as *const esp_task_wdt_config_t) })?;
```

Comment:
Not needed if upper cfg is correct.

Reply:
You are right. That's why I have made a similar comment in the related issue. I can remove the line and assume that the upper cfg is correct.

Reply:
Not sure that I understand why it is not needed. esp_task_wdt_init() needs to be called. Between v4.4.3 and v5.0 the signature of the function has changed. If I remove the line, esp_task_wdt_init() will not be called. The upper cfg will disable the following if structure in case of IDF v5.0 and therefor esp_task_wdt_init() will not be called line 479 will also be disabled as it is only enabled in case of IDF v4.

Reply:
The upper `cfg` is enabled from 5.0 and upper. The `init_by_idf ` is true if the watchdog was enabled by `esp-idf`. This is the case if watchdog is enabled in the `sdkconfig`. Until version 4.x, `esp_task_wdt_init()` could be called if the `twdt` already initialized. But with  `esp_task_wdt_reconfigure()` `esp_task_wdt_init()` will result in an error if `twdt` already initialized.

Reply:
So let me double check on my side regarding ESP_IDF installation - we don't need to fix something which is not broken.

Reply:
Okay I removed esp support and re-installed:
```espup install --esp-idf-version release/v5.0```
Now it compiled without the error, so I'm assuming I'm using now the same toolchain / IDF version as the CI build. Previously I had installed the toolchain with:
```espup install --esp-idf-version v5.0```
I was assuming this was the stable one - might be, as it gave the error.
So my understanding is now that CI compiles against unstable release/v5.0 and not stable v5.0 which is why the CI build doesn't show the error. Would be good if someone can confirm the differences between the different IDF v5.0 versions and than check if it is intentional that the CI build compiles against unstable v5 (if my understanding is correct).